AdvancedTeleport-Bukkit/src/main/java/io/github/niestrat99/advancedteleport/commands/home/HomesCommand.java

```diff
@@ -95,77 +95,85 @@ public boolean onCommand(
+                                        atPlayer.canAccessHome(home)
+                                                || ExPermission
+                                                .hasPermissionOrStar(
+                                                        sender,
+                                                        "at.admin.homes")
+                                }) // How the fuck do you associate a value
+                        // like a pair in java?
+                        .map(
+                                pair -> {
+                                    final var home = (Home) pair[0];
+                                    final var canAccess = (boolean) pair[1];
+                                    final var baseComponent =
+                                            Component.text(home.getName())
+                                                    .hoverEvent(
+                                                            CustomMessages
+                                                                    .locationBasedTooltip(
+                                                                            sender,
+                                                                            home,
+                                                                            "homes"));
+
+                                    CoreClass.debug("Processing home: " + home.getName() + ", " + canAccess);
+
+                                    if (!canAccess) {
+                                        if (!MainConfig.get()
+                                                .HIDE_HOMES_IF_DENIED
+                                                .get())
+                                            return Component
+                                                    .empty(); // TODO: Make sure
+                                        // this doesn't cause
+                                        // an extra comma.
+                                        return baseComponent
+                                                .color(NamedTextColor.GRAY)
+                                                .decorate(TextDecoration.ITALIC);
+                                    }
+
+                                    return baseComponent.clickEvent(
+                                            ClickEvent.runCommand(
+                                                    "/advancedteleport:home "
+                                                            + (sender == target
+                                                            ? ""
+                                                            : target
+                                                            .getName()
+                                                            + " ")
+                                                            + home.getName()));
+                                })
+                        .toList());
+        CoreClass.debug("Message text built.");
```

Comment:
With the change in how homes are checked, we can move the variable back to where it was prior so that we don't waste server processing on it if it's not used.